### PR TITLE
LibCore: Explain EventLoop and reorder some members in the header

### PR DESCRIPTION
**Pull request by kleinesfilmroellchen**
This hopefully makes EventLoop easier to understand.

### CODE REVIEW

**Review by GMTA — APPROVED** [review]
LGTM, this definitely would have helped my understanding of EventLoop when I first encountered it.

**Comment by kleinesfilmroellchen** [comment]
@GMTA can you not merge this if you think it looks good?

**Pull request merged**
